send_conn.go

```diff
@@ -13,22 +13,56 @@ type sendConn interface {
 }
 
 type sconn struct {
-	net.PacketConn
+	connection
 
 	remoteAddr net.Addr
+	info       *packetInfo
 }
 
 var _ sendConn = &sconn{}
 
-func newSendConn(c net.PacketConn, remote net.Addr) sendConn {
-	return &sconn{PacketConn: c, remoteAddr: remote}
+func newSendConn(c connection, remote net.Addr, info *packetInfo) sendConn {
+	return &sconn{connection: c, remoteAddr: remote, info: info}
 }
 
 func (c *sconn) Write(p []byte) error {
-	_, err := c.PacketConn.WriteTo(p, c.remoteAddr)
+	_, err := c.WritePacket(p, c.remoteAddr, c.info)
```

Comment:
Would it make sense to serialize the `packetInfo` once, instead of doing it for every `WritePacket` call? This would save a lot of garbage, wouldn't it?

Reply:
Yes, that's a good point.

Reply:
We also probably don't need all this machinery in case the server does not listen on ANY.

On freebsd, there is also the limitation that it does not seem to work if ANY v4+v6 is bound, you need to create v4 and v6 listeners separately.